### PR TITLE
Add PostObject Checksum

### PR DESCRIPTION
**Pull request by klauspost**
## Description

This adds checksums as described in https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html

## How to test this PR?

Mint test PR: https://github.com/minio/minio-go/pull/1827

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by klauspost** [comment]
(should be trivial to merge with https://github.com/minio/minio/pull/17233 )

File: cmd/bucket-handlers.go [hunk 1]
```diff
@@ -1018,6 +1018,18 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 		writeErrorResponse(ctx, w, apiErr, r.URL)
 		return
 	}
+	checksum, err := hash.GetContentChecksum(formValues)
+	if err != nil {
+		apiErr := errorCodes.ToAPIErr(ErrMalformedPOSTRequest)
+		apiErr.Description = fmt.Sprintf("%s (%v)", apiErr.Description, fmt.Errorf("Invalid checksum: %w", err))
+		writeErrorResponse(ctx, w, apiErr, r.URL)
```
Comment (harshavardhana) [inline cmd/bucket-handlers.go:1025]: missing return

File: cmd/bucket-handlers.go [hunk 2]
```diff
@@ -1018,6 +1018,18 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
...
+	}
+	if checksum != nil && checksum.Type.Trailing() {
+		// Not officially supported in POST requests.
+		apiErr := errorCodes.ToAPIErr(ErrMalformedPOSTRequest)
+		apiErr.Description = fmt.Sprintf("%s (%v)", apiErr.Description, errors.New("Trailing checksums not available for POST operations"))
+		writeErrorResponse(ctx, w, apiErr, r.URL)
```
Comment (harshavardhana) [inline cmd/bucket-handlers.go:1032]: missing return

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**